app/models/shipping_info.rb

```diff
@@ -0,0 +1,60 @@
+class ShippingInfo
+  def initialize(params)
+    @params = params
+  end
+
+  def ups_rates
+    ups = ActiveShipping::UPS.new(login: ENV["UPS_LOGIN"], password: ENV["UPS_PASSWORD"], key: ENV['UPS_KEY'])
+    ups_response = ups.find_rates(origin, destination, packages)
+    ups_response.rates.sort_by(&:price).collect {|rate| [rate.service_name, rate.price]}
+  end
+
+  def usps_rates
+    usps = ActiveShipping::USPS.new(login: ENV['USPS_LOGIN'])
+    usps_response = usps.find_rates(origin, destination, packages)
+    usps_response.rates.sort_by(&:price).collect {|rate| [rate.service_name, rate.price]}
+  end
+
+  private
+
+  def origin
+    @origin ||= ActiveShipping::Location.new(country: "US",
+      state: @params[:order][:origin][:state],
+      city: @params[:order][:origin][:city],
+      zip: @params[:order][:origin][:zip])
+  end
+
+  def destination
+    @destination ||= ActiveShipping::Location.new(country: 'US',
```

Comment:
do `@destination` and `@origin` and `@packages` (below) need to be set as instance variables? Since they're not being used anywhere else, could they just be local variables?